lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4367,6 +4373,34 @@ static bool checkTransposedParameters(
 }
 
 // SWIFT_ENABLE_TENSORFLOW
+// Determine whether the 'static' 'self' type of the transpose method is the
+// same as the one in the result. This should only be called when a linearity
+// parameter is 'self'.
+static bool transposeSelfTypesMatch(
```

Comment:
Please document the function’s behavior as to what it does to arguments ‘staticSelfType’ and ‘initSelfType’. Also, it is more common to take output variables as references (&) instead of pointers.